panoptes_client/tests/test_workflow.py

```diff
@@ -14,18 +15,35 @@ class TestWorkflow(unittest.TestCase):
     def setUp(self):
         super().setUp()
         caesar_post_patch = patch.object(Caesar, 'http_post')
+        caesar_put_patch = patch.object(Caesar, 'http_put')
         caesar_get_patch = patch.object(Caesar, 'http_get')
         self.caesar_post_mock = caesar_post_patch.start()
+        self.caesar_put_mock = caesar_put_patch.start()
         self.caesar_get_mock = caesar_get_patch.start()
         self.addCleanup(caesar_post_patch.stop)
         self.addCleanup(caesar_get_patch.stop)
+        self.addCleanup(caesar_put_patch.stop)
 
-    def test_add_to_caesar(self):
+    def test_save_to_caesar_update(self):
         workflow = Workflow(1)
-        workflow.add_to_caesar()
+        workflow.save_to_caesar()
 
-        self.caesar_post_mock.assert_called_once()
+        self.caesar_put_mock.assert_called_once()
+        self.caesar_put_mock.assert_called_with('workflows/1', json={
+            'workflow': {
+                'id': workflow.id,
+                'public_extracts': False,
+                'public_reductions': False
+            }
+        })
+
+    def test_save_to_caesar_create(self):
+        self.caesar_get_mock.side_effect = PanoptesAPIException("Couldn't find Workflow with 'id'=1")
```

Comment:
is this the actual error string?  i think it might be `Could not` which may impact the way the current 404 error handling code works
```
$ panoptes workflow info 1000000001
.....
/panoptes.py", line 620, in http_get
    return Panoptes.client().get(
  File "/usr/local/lib/python3.9/site-packages/panoptes_client/panoptes.py", line 316, in get
    return self.json_request(
  File "/usr/local/lib/python3.9/site-packages/panoptes_client/panoptes.py", line 281, in json_request
    raise PanoptesAPIException(', '.join(
panoptes_client.panoptes.PanoptesAPIException: Could not find workflow with id='1000000001'
```

Reply:
i believe it is https://caesar-staging.zooniverse.org/workflows/1, but maybe we could check for both `couldn't` and `could not`?

Reply:
right, if it means anything, Caesar does return a 404, but the way python client deals with error responses (or at least the way I have it,) which reuses the `Panoptes.json_response` method, which raises anything not a success status with `PanoptesAPIException` and error message, so we do lose the status on the way. https://github.com/zooniverse/panoptes-python-client/blob/437b6f3eed94000acfb0250e0e7cd14ac99a2fed/panoptes_client/panoptes.py#L282

i guess it could be updated to send status or make `json_response` more generic, but possibly out of scope for now. 

appreciate the review Cam. 👍